### PR TITLE
Typing Indicator

### LINKED ISSUE #72: Show when a user is typing

### PR DESCRIPTION
**Pull request by TheConnMan**
Closes #72

### CODE REVIEW

**Comment by SirZach** [comment]
https://docs.angularjs.org/guide/component Their docs don't use scope in components

**Comment by SirZach** [comment]
http://stackoverflow.com/questions/16619740/angular-should-i-use-this-or-scope Controller as seems to be more of a thing with future angular apps

**Comment by TheConnMan** [comment]
Very good to know, thanks for doing some digging. I'll convert back to using the controller variable. I'll also bind to the keydown event so I don't have to watch that variable.

**Comment by SirZach** [comment]
@TheConnMan I've also discovered angular has a `ng-change` directive which I think will fire a function to simulate the `$watch`

File: assets/js/components/chat.js [hunk 1]
```diff
@@ -56,35 +60,33 @@ function ChatController($scope, $http, $notification, $storage) {
...
     io.socket.on('typers', (typers) => {
-      if (typers.indexOf(self.username) !== -1) {
-        typers.splice(typers.indexOf(self.username), 1);
+      if (typers.indexOf(self.getUsername()) !== -1) {
+        typers.splice(typers.indexOf(self.getUsername()), 1);
       }
       if (typers.length === 0) {
-        $scope.typers = '';
+        self.typers = '';
       } else if (typers.length === 1) {
```
Comment (SirZach) [inline assets/js/components/chat.js]: Shudders at using closures. I'll update it later

**Pull request merged**